tests/test_sensor.py

```diff
@@ -260,6 +260,17 @@ async def test_time_charge_complete_charging(
 
     assert state.state == charge_complete_str
 
+    car_mock_data.VEHICLE_DATA["charge_state"]["time_to_full_charge"] = 0.15
+    earlier_charge_complete = mock_now + timedelta(
+        hours=float(car_mock_data.VEHICLE_DATA["charge_state"]["time_to_full_charge"])
+    )
+    earlier_charge_complete_str = datetime.strftime(
+        earlier_charge_complete, "%Y-%m-%dT%H:%M:%S+00:00"
+    )
+
+    state = hass.states.get("sensor.my_model_s_time_charge_complete")
```

Comment:
This is pulling the same state from line 261. Maybe HA does not update the state between the lines or I'm not changing the correct values for HA to set a new state.